### DISCUSSION TITLE
write some Data on disk during an async function

### DISCUSSION THREAD

**Discussion by ChristopheTdn** [Rapptz/discord.py#8278] [Q&A] [answered]
hello ! I need to write some Data during  an async function but in this sample the script refuse to write the variable value (`counter`) on disk. Curiously if you write only texte , it works (` f.write("only text...")`)...  I need this function to write important variable on disk. Somenone can help me ?
```python
import discord
import asyncio

class MyClient(discord.Client):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        # create the background task and run it in the background
        self.bg_task = self.loop.create_task(self.my_background_task())
    async def on_ready(self):
        print('Logged in as')
        print(self.user.name)
        print(self.user.id)
        print('------')
    async def my_background_task(self):
        await self.wait_until_ready()
        counter = 0
        channel = self.get_channel(979869063905939456) # channel ID goes here
        while not self.is_closed():
            counter += 1
            with open('readme.txt', 'w') as f:
                f.write(counter) 
            await channel.send(counter)
            await asyncio.sleep(2) # task runs every 60 seconds

intents = discord.Intents.default()
intents.messages = True
client = MyClient(intents=intents)
client.run('tokenhere')```

**Comment by LostLuma** [comment, marked as answer]
Your background task is running into an error because you can not pass an integer to the `write` function:
```py
>>> with open('example.txt', 'w') as f:
...     f.write(1)
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
TypeError: write() argument must be str, not int
```
You will have to convert your counter variable to a string when writing it to the file instead.

To see this sort of error you might want to wrap your code in a try except block to be able to log it, or you can base your code off of the other [background task example](https://github.com/Rapptz/discord.py/blob/master/examples/background_task.py) using the tasks extension, as it will print/log errors by default.
